lib/modules/datasource/unity3d/index.spec.ts

```diff
@@ -0,0 +1,114 @@
+import { getPkgReleases } from '..';
+import { Fixtures } from '../../../../test/fixtures';
+import * as httpMock from '../../../../test/http-mock';
+import { Unity3dDatasource } from '.';
+
+const mockRSSFeeds = (streams: { [keys: string]: string }) => {
+  Object.entries(streams).map(([stream, url]) => {
+    const content = Fixtures.get(stream + '.xml');
+
+    const uri = new URL(url);
+
+    httpMock.scope(uri.origin).get(uri.pathname).reply(200, content);
+  });
+};
+
+describe('modules/datasource/unity3d/index', () => {
+  it('returns stable and lts releases by default', async () => {
+    const qualifyingStreams = { ...Unity3dDatasource.streams };
+    delete qualifyingStreams.beta;
+    mockRSSFeeds(qualifyingStreams);
+    const responses = (
+      await getPkgReleases({
+        datasource: Unity3dDatasource.id,
+        packageName: 'm_EditorVersion',
+      })
+    )?.releases.map((release) => release.version);
+
+    // check that only letter f is present (final releases)
+    expect(responses!.every((version) => /[fp]/.test(version))).toBe(true);
+    // check that no version contains the letter b (beta release)
+    expect(responses!.every((version) => !/b/.test(version))).toBe(true);
```

Comment:
Rather use toMatchObject or toEqual. That way there is less logic in tests.

Reply:
Was not aware of how powerful the selectors in Jest can be; thanks for pointing that out, learned a ton!

I've reworked most tests inside 0855e41dbac3e32d1ff6e5c42519b90c8c687c7c and tried to keep every commit since your review atomic. You should be able to look at each commit individually if looking at a diff between "before and after the review" becomes too much.

---

I've attempted to rewrite the `returns different versions for each stream` test to work with the Jest `…containing()`-methods, but couldn't find any readable way to satisfy both the `some items in responses.stable are in responses.lts` but `not all items in responses.stable are in responses.lts` conditions.  
Open to suggestions, if there is a better way that I'm missing.